solidity/contracts/integrator/TBTCDepositorProxy.sol

```diff
@@ -0,0 +1,309 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity ^0.8.0;
+
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+
+import "./IBridge.sol";
+import "./ITBTCVault.sol";
+
+/// @title Abstract TBTCDepositorProxy contract.
+/// @notice This abstract contract is meant to facilitate integration of protocols
+///         aiming to use tBTC as an underlying Bitcoin bridge.
+///
+///         Such an integrator is supposed to:
+///         - Create a child contract inheriting from this abstract contract
+///         - Call the `__TBTCDepositorProxy_initialize` initializer function
+///         - Use the `_initializeDeposit` and `_finalizeDeposit` as part of their
+///           business logic in order to initialize and finalize deposits.
+///
+/// @dev Example usage:
+///      ```
+///      // Example upgradeable integrator contract.
+///      contract ExampleTBTCIntegrator is TBTCDepositorProxy, Initializable {
+///          /// @custom:oz-upgrades-unsafe-allow constructor
+///          constructor() {
+///              // Prevents the contract from being initialized again.
+///              _disableInitializers();
+///          }
+///
+///          function initialize(
+///              address _bridge,
+///              address _tbtcVault
+///          ) external initializer {
+///              __TBTCDepositorProxy_initialize(_bridge, _tbtcVault);
+///          }
+///
+///          function startProcess(
+///              IBridgeTypes.BitcoinTxInfo calldata fundingTx,
+///              IBridgeTypes.DepositRevealInfo calldata reveal
+///          ) external {
+///              // Embed necessary context as extra data.
+///              bytes32 extraData = ...;
+///
+///              uint256 depositKey = _initializeDeposit(
+///                  fundingTx,
+///                  reveal,
+///                  extraData
+///              );
+///
+///              // Use the depositKey to track the process.
+///          }
+///
+///          function finalizeProcess(uint256 depositKey) external {
+///              (uint256 tbtcAmount, bytes32 extraData) = _finalizeDeposit(depositKey);
+///
+///              // Do something with the minted TBTC using context
+///              // embedded in the extraData.
+///          }
+///      }
+abstract contract TBTCDepositorProxy {
```

Comment:
Why not call it just `TBTCDepositor` or `AbstractTBTCDepositor`? The proxy suffix could suggest it is some sort of an upgradeable proxy contract.

Reply:
I wanted to avoid pure "depositor" to not mix it with the concept of "depositor" already present in the `Bridge`. I don't have strong preferences here though. `AbstractTBTCDepositor` is completely fine.

Reply:
See 68d02e86